pypred/ast.py

```diff
@@ -307,38 +307,96 @@ def failure_info(self, ctx):
 
 
 class ContainsOperator(Node):
-    "Used for the 'contains' operator"
-    def __init__(self, left, right):
+    "Used for the 'contains' operator and derived list comparisons"
+
+    def __init__(self, op, left, right):
+        self.type = op
         self.left = left
         self.right = right
 
+    def name(self):
+        return "%s operator at %s" % (self.type.upper(), self.position)
+
     def _validate(self, info):
-        if not isinstance(self.right, (Number, Literal, Constant)):
+        if self.type not in ("contains", "any", "all"):
```

Comment:
Does pypred have any constants/enums to declare types? It seems odd/error prone to use the string everywhere.

Reply:
From what I've seen, no. I agree it's pretty bad and error prone, but I tried to stay in line with what was there. If we don't want to merge back to the original project, I have a lot of improvements in mind that can be done 😁

Reply:
Haha fair, no this is fine. I agree to keep the `pypred` style as much as possible unless we don't get any feedback from the maintainers.